src/model.js

```diff
@@ -1058,37 +1059,49 @@ var Model = TransisObject.extend(function() {
   };
 
   // Public: Runs registered validators for the given attribute. This will clear any existing
-  // validation errors for the given attribute.
+  // validation errors for the given attribute. The registered validators can be reduced by
+  // providing an optional context.
   //
   // name - The name of the attribute to run validations for.
+  // ctx  - The context in which to run validations on the given attribute. (optional)
   //
   // Returns `true` if no validation errors are found on the given attribute and `false` otherwise.
-  this.prototype.validateAttr = function(name) {
+  this.prototype.validateAttr = function(name, ctx) {
     this._clearErrors(name);
-
     if (!this.validators[name]) { return true; }
 
-    for (let i = 0, n = this.validators[name].length; i < n; i++) {
-      let validator = this.validators[name][i];
-      if (typeof validator === 'function') { validator.call(this); }
-      else if (typeof validator === 'string' && validator in this) { this[validator](); }
-      else { throw new Error(`${this.constructor}#validateAttr: don't know how to execute validator: \`${validator}\``); }
+    let attrCtx = ctx && this.validators[name].find( (v)=> {return v.opts.on && v.opts.on === ctx} );
+    if(attrCtx) {
+      attrCtx.validator.call(this);
```

Comment:
This won't work if I specify a context and a method name instead of a function:

```javascript
this.validate('foo', 'validateFoo', {on: 'something'});

this.prototype.validateFoo = function() {
};
```